ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/tasks/DownloadHeaderSequenceTask.java

```diff
@@ -195,16 +199,40 @@ public static DownloadHeaderSequenceTask endingAtHeader(
               child =
                   (headerIndex == segmentLength - 1) ? referenceHeader : headers[headerIndex + 1];
             }
+            final ProtocolSpec protocolSpec = protocolSchedule.getByBlockNumber(child.getNumber());
+            final BadBlockManager badBlockManager = protocolSpec.getBadBlocksManager();
 
             if (!validateHeader(child, header)) {
               // Invalid headers - disconnect from peer
-              LOG.debug(
-                  "Received invalid headers from peer, disconnecting from: {}",
-                  headersResult.getPeer());
-              headersResult.getPeer().disconnect(DisconnectReason.BREACH_OF_PROTOCOL);
-              future.completeExceptionally(
-                  new InvalidBlockException(
-                      "Header failed validation.", child.getNumber(), child.getHash()));
+
+              final BlockHeader invalidBlock = child;
```

Comment:
Add some comments explaining that even though the header is known bad we are downloading the block body for the debug_badBlocks RPC.

Reply:
Done